lib/asset_sync/storage.rb

```diff
@@ -4,7 +4,7 @@
 
 module AssetSync
   class Storage
-    REGEXP_FINGERPRINTED_FILES = /^(.*)\/([^-]+)-[^\.]+\.([^\.]+)$/
+    REGEXP_FINGERPRINTED_FILES = /\A(.*)\/(.+)-[^\.]+\.([^\.]+)\z/
```

Comment:
Any reason you changed the anchors as well?

[Ref for anchors](https://ruby-doc.org/core-3.0.0/Regexp.html#class-Regexp-label-Anchors)

Reply:
The original anchors only anchored per line and not the whole string. While rare, it is possible to have file names with new lines.

Reply:
Is it possible to add a test case for file name with new line then?

Reply:
Added. I originally left it out since it's such an extreme edge case, though it did flush out another issue.